### ISSUE TITLE
When the ml decoder is applied to my custom data, the loss does not decrease during training.

### ISSUE THREAD

**Issue by jeongHwarr** [Alibaba-MIIL/ML_Decoder#6]
Thank you for your impressive project. 

When I using the ml decoder, training is not working properly.
(Train loss doesn't decrease, accuracy also does not improve.)

I used the resnet50 model pretrained with Imagenet 21 provided by your group. 

Backbone model was created by `timm` library.
```
 model = timm.create_model('resnet50', pretrained=False, num_classes=0)
 model = load_model_weights(model, os.path.join(os.path.split(os.path.realpath(__file__))[0],'resnet50_miil_21k.pth'))
```

This is my Classifier. 

```
class Classifier(nn.Module):
    def __init__(self, cfg):
        super(Classifier, self).__init__()
        self.cfg = cfg
        self.backbone = BACKBONES[cfg.backbone](cfg)
        self.num_in_channel = self.get_num_channel() #number of output channels of backbone model
        self.ml_decoder_head = MLDecoder(self.cfg.num_class, initial_num_features=self.num_in_channel)

    def forward(self, x):
        # (N, C, H, W)
        x = self.backbone(x)
        x = self.ml_decoder_head(x)
        return x
```
For logit `x`, `nn.CrossEntropyLoss` was applied.

When I used the head by applying GAP and adding a linear layer, this issue did not occur. 

The number of classes is 10, and it is a single label.

**Comment by mrT23** [comment]
i think you are not giving MLDecoder the correct input. the inputs it expects is the spatial embedding tensor, before the GAP pooling (7x7x2048). Triple check for bugs in your implementation.

as a sanity test, try to apply your model on CIFAR10\100. if you encounter problems there, it is indeed a bug, since ML-Decoder works well [there](https://paperswithcode.com/sota/image-classification-on-cifar-100?p=ml-decoder-scalable-and-versatile)

This code demonstrates how to properly use ResNet50 with MLDecoder:

```
    model = timm.create_model('resnet50', pretrained=False, num_classes=10)
    model.global_pool = torch.nn.Identity()
    model.fc = MLDecoder(num_classes=10, initial_num_features=model.num_features)
    ...
    logits=model(x)
```

**Comment by jeongHwarr** [comment]
@mrT23  Thank you for your reply! The shape of my input is (batch size, num_channel, height, width). I think there is nothing wrong with the input. Am I right? 

By reducing the learning rate, trainining works well!
It seems that the learning rate should have been smaller than when applying GAP. I'll  also test it with your code. Thank you.

**Comment by mrT23** [comment]
ok. as a rule of thumb, Adam optimizer with lr 1e-4 to 5e-4 works well

good luck
